bridge/client/app/_components/ktb-create-secret-form/ktb-create-secret-form.component.ts

```diff
@@ -0,0 +1,52 @@
+import {Component, OnInit} from '@angular/core';
+import {DataService} from '../../_services/data.service';
+import {Router} from '@angular/router';
+import {FormControl, FormGroup, Validators} from "@angular/forms";
+import {Secret} from "../../_models/secret";
+
+@Component({
+  selector: 'ktb-secrets-view',
+  templateUrl: './ktb-create-secret-form.component.html',
+  styleUrls: ['./ktb-create-secret-form.component.scss']
+})
+export class KtbCreateSecretFormComponent implements OnInit {
+
+  public isLoading: Boolean = false;
+  public secret: Secret = null;
+
+  public defaultFormControls: {} = {
+    name: new FormControl('', [Validators.required])
+  };
+  public createSecretForm = new FormGroup(this.defaultFormControls);
+
+  constructor(private dataService: DataService, private router: Router) { }
+
+  ngOnInit(): void {
+    this.secret = new Secret();
+    this.addPair();
+  }
+
+  createSecret() {
+    if(this.createSecretForm.valid) {
+      this.isLoading = true;
+      this.dataService.addSecret(this.secret)
+        .subscribe((result) => {
+          this.isLoading = false;
+          this.router.navigate(['../secrets']);
```

Comment:
This redirects to `http://.../`
Change it to `this.router.navigate(['../'],  { relativeTo: this.route });` (`this.route` is ActivatedRoute)